C7FIT/MapViewController.swift

```diff
@@ -0,0 +1,172 @@
+        timer.invalidate()
+        locationManager.stopUpdatingLocation()
+    }
+    
+    // MARK: Create Run Data
+    func createRunData() {
+        var tempRun = RunData()
+        var savedLocations = [Location]()
+        
+        print(savedLocations)
+        var i = 0
+        for location in locations {
+            var tempLocation = Location()
+            tempLocation.timestamp = location.timestamp.description
+            tempLocation.latitude = location.coordinate.latitude
+            tempLocation.longitude = location.coordinate.longitude
+            savedLocations.append(tempLocation)
+            i = i + 1
+        }
+        tempRun.distance = distance
+        tempRun.time = seconds
+        tempRun.pace = pace
+        tempRun.locations = savedLocations
+        lastRun = tempRun
+    }
+    
+    // MARK: location tracking
+    func setupLocationTracking() {
+        locationManager.delegate = self
+        locationManager.desiredAccuracy = kCLLocationAccuracyNearestTenMeters
+        locationManager.activityType = .fitness
+        locationManager.distanceFilter = 5
+        locationManager.requestAlwaysAuthorization()
+        if (CLLocationManager.locationServicesEnabled()) {
+            locationManager.startUpdatingLocation()
+        }
+    }
+    
+    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+        for location in locations {
+            //Focus on the runner
+            let center = CLLocationCoordinate2D(latitude: location.coordinate.latitude, longitude: location.coordinate.longitude)
+            let region = MKCoordinateRegion(center: center, span: MKCoordinateSpan(latitudeDelta: 0.01, longitudeDelta: 0.01))
+            mapViewApp.mapView.setRegion(region, animated: true)
+            
+            if location.horizontalAccuracy < 50 {
+                if self.locations.count > 0 {
+                    distance += location.distance(from: self.locations.last!)
+                }
+                self.locations.append(location)
+            }
+        }
+    }
+    
+    func locationManager(_ manager: CLLocationManager, didFailWithError error: Error)
+    {
+        print("Error \(error)")
+    }
+    
+    // MARK: Timer
+    func eachSecond(timer: Timer) {
+        seconds = seconds + 1
+//        let totalTimeQuantity = seconds
+        let minuteQuantity = HKQuantity(unit: HKUnit.minute(), doubleValue: floor(seconds/60))
+        let secondsQuantity = HKQuantity(unit: HKUnit.second(), doubleValue: seconds.truncatingRemainder(dividingBy: 60.0))
+        mapViewApp.secondsQuantity.text = "Time Elapsed: " + minuteQuantity.description + " " + secondsQuantity.description
+        
+        let distanceQuantity = HKQuantity(unit: HKUnit.meter(), doubleValue: distance)
+        mapViewApp.distanceQuantity.text = "Distance: " + distanceQuantity.description
+        
+    // convert to mile per minute
```

Comment:
does this code really convert to miles per minute? would not guess that by looking at it